force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,193 @@
+            //System.debug(opportunitiesToUpdate);
+           
+        }
+        List<Opportunity> bRecords = [SELECT Id, Primary_Contact__c FROM Opportunity WHERE AccountId IN :opportunitiesWithoutPrimaryContactToAccountIdMap.values() LIMIT 5];    
+        try{
+            update opportunitiesToUpdate;
+        }catch (DmlException e) {
+            // Output the error message to the debug log
+            System.debug('An error occurred: ' + e.getMessage());
+            // Optionally, you can output more details such as:
+            System.debug('Error status code: ' + e.getDmlStatusCode(0));
+            System.debug('Error field: ' + e.getDmlFields(0));
+            
+            System.debug('Error stack trace: ' + e.getStackTraceString());
+        }
+    
+        update opportunitiesToUpdate;
+        List<Opportunity> aRecords = [SELECT Id, Primary_Contact__c FROM Opportunity WHERE AccountId IN :opportunitiesWithoutPrimaryContactToAccountIdMap.values() LIMIT 5];
+    }
+
+    public void setDefaultOpportunityType(){
+        // Set default Type for new Opportunities
+        for(Opportunity opp : triggerList){
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+            }
+        }
+    }
+
+    /*
+    * deleteCloseWonOpportunity()
+    * When an opportunity is deleted prevent the deletion of a closed won opportunity if the account industry is 'Banking'.
+    * Trigger should only fire on delete.
+    */
+    public void deleteCloseWonOpportunity(){
+        List<Id> accIds = new List<Id>();
+        for(Opportunity opp : triggerList){
+            accIds.add(opp.AccountId);
+        }
+        List<Account> relatedAccounts = [SELECT Id, Industry FROM Account WHERE Id IN :accIds];
+
+        Map<Id, String> accIdIndustryMap = new Map<Id, String>();
+
+        for(Account acc : relatedAccounts){
+            accIdIndustryMap.put(acc.Id, acc.Industry);
+        }
+        
+        for(Opportunity opp : triggerList){
+            if( accIdIndustryMap.get(opp.AccountId) == 'Banking' && opp.StageName == 'Closed Won'){
+                opp.addError('Cannot delete closed opportunity');
+            }
+        }
+    }
+
+    // Create a new Task for newly inserted Opportunities
+    public void insertTask(){
+        List<Task> tasksToInsert = new List<Task>();
+        for (Opportunity opp : triggerList){
+            Task tsk = new Task();
+            tsk.Subject = 'Call Primary Contact';
+            tsk.WhatId = opp.Id;
+            tsk.WhoId = opp.Primary_Contact__c;
+            tsk.OwnerId = opp.OwnerId;
+            tsk.ActivityDate = Date.today().addDays(3);
+            tasksToInsert.add(tsk);
+        }
+        insert tasksToInsert;
+    }
+
+    /*
+    * setPrimaryContact()
+    * Before Update
+    * When an opportunity is updated set the primary contact on the opportunity to the contact with the title of 'CEO'.
+    */
+    public void setPrimaryContact(){
```

Comment:
Is there a big difference between this method and assignPrimaryContact method? Should those maybe be combined into one method?